tests/test_exp_attn_res.py

```
@@ -0,0 +1,232 @@
+        for name, parameter in checkpointed.named_parameters()
+        if parameter.grad is not None
+    }
+
+    assert reference_grads.keys() == checkpointed_grads.keys()
+    for name, gradient in reference_grads.items():
+        torch.testing.assert_close(
+            gradient,
+            checkpointed_grads[name],
+            rtol=0,
+            atol=0,
+            msg=lambda message: f"{name}: {message}",
```

Ruff (0.16.0)

[warning] Function definition does not bind loop variable `name`

(B023)